src/main/scala/io.circe.config/parser.scala

```diff
@@ -105,26 +119,180 @@ object parser extends Parser {
+   * scala> case class AppSettings(http: HttpSettings)
+   *
+   * scala> parser.decodeFile[AppSettings](new java.io.File("src/test/resources/application.conf"))
+   * res0: Either[io.circe.Error, AppSettings] = Right(AppSettings(HttpSettings(ServerSettings(localhost,8080))))
+   * }}}
+   */
   final def decodeFile[A: Decoder](file: File): Either[Error, A] =
-    finishDecode[A](parseFile(file))
-
-  final def decodeAccumulating[A: Decoder](config: Config): ValidatedNel[Error, A] =
-    finishDecodeAccumulating[A](parse(config))
+    finishDecode(parseFile(file))
 
-  final def decodePath[A: Decoder](path: String): Either[Throwable, A] =
-    Either.catchNonFatal(ConfigFactory.load()).flatMap(decodePath[A](_, path).leftWiden[Throwable])
+  /**
+   * Load the default configuration and decode an instance.
+   *
+   * @example
+   * {{{
+   * scala> import io.circe.generic.auto._
+   * scala> case class ServerSettings(host: String, port: Int)
+   *
+   * scala> parser.decodePath[ServerSettings]("http.server")
+   * res0: Either[io.circe.Error, ServerSettings] = Right(ServerSettings(localhost,8080))
+   * }}}
+   */
+  final def decodePath[A: Decoder](path: String): Either[Error, A] =
+    finishDecode(parsePath(path))
 
+  /**
+   * Decode of an instance at a specific path.
+   *
+   * @example
+   * {{{
+   * scala> import io.circe.generic.auto._
+   * scala> case class ServerSettings(host: String, port: Int)
+   *
+   * scala> import com.typesafe.config.ConfigFactory
+   * scala> val config = ConfigFactory.load()
+   *
+   * scala> parser.decodePath[ServerSettings](config, "http.server")
+   * res0: Either[io.circe.Error, ServerSettings] = Right(ServerSettings(localhost,8080))
+   * }}}
+   */
   final def decodePath[A: Decoder](config: Config, path: String): Either[Error, A] =
-    if (config.hasPath(path)) decode[A](config.getConfig(path))
-    else Left(ParsingFailure("Path not found in config", new ConfigException.Missing(path)))
+    finishDecode(parsePath(config, path))
+
+
+  final def decodeAccumulating[A: Decoder](config: Config): ValidatedNel[Error, A] =
+    finishDecodeAccumulating[A](parse(config))
 
   final def decodeFileAccumulating[A: Decoder](file: File): ValidatedNel[Error, A] =
     finishDecodeAccumulating[A](parseFile(file))
 
+
+  /**
+   * Load default configuration and decode an instance supporting [[cats.ApplicativeError]].
+   *
+   * @example
+   * {{{
+   * scala> import io.circe.generic.auto._
+   * scala> case class ServerSettings(host: String, port: Int)
+   * scala> case class HttpSettings(server: ServerSettings)
+   * scala> case class AppSettings(http: HttpSettings)
+   *
+   * scala> import cats.effect.IO
+   * scala> parser.decode[IO, AppSettings]()
+   * res0: cats.effect.IO[AppSettings] = IO(AppSettings(HttpSettings(ServerSettings(localhost,8080))))
+   * }}}
+   */
+  final def decode[F[_], A: Decoder]()(implicit ev: ApplicativeError[F, Throwable]): F[A] =
```

Comment:
Overloads with different numbers of type parameters AND different return types will require explicit type parameters, see PR comment

Reply:
So maybe just this would be enough:

```
def decode[A: Decoder](): Either[Error, A]
def decode[A: Decoder](c : Config): Either[Error, A]
def decodeF[F[_]: TErr, A: Decoder](): F[A]
def decodeF[F[_]: TErr, A: Decoder](c : Config): F[A]
```

Reply:
Thanks for the review. Will do, however, the type parameters are required even with specific name:

```scala
val settings: IO[AppSettings] = parser.decodeF()
```
gives
```
[error] /opt/circe-config/target/scala-2.13.0-M5/src_managed/test/READMEmd2Doctest.scala:81:51: diverging implicit expansion for type io.circe.Decoder[A]
[error] starting with method decodeNonEmptyChain in object Decoder
[error]     val settings: IO[AppSettings] = parser.decodeF()
```